nullaway/src/test/java/com/uber/nullaway/jspecify/GenericMethodTests.java

```diff
@@ -161,6 +161,234 @@ public void genericMethodAndVoidTypeWithInference() {
+            "        // legal",
+            "        Foo<@Nullable Object> f1 = Foo.makeNull(null);",
+            "        // BUG: Diagnostic contains: passing @Nullable parameter 'null' where @NonNull is required",
+            "        Foo<Object> f2 = Foo.makeNull(null);",
+            "        Foo<@Nullable Object> f3 = Foo.makeNull(new Object());",
+            "        Foo<Object> f4 = Foo.makeNull(new Object());",
+            "        // ILLEGAL: U does not have a @Nullable upper bound",
+            "        // BUG: Diagnostic contains: passing @Nullable parameter 'null' where @NonNull is required",
+            "        Foo<@Nullable Object> f5 = Foo.makeNonNull(null);",
+            "        // BUG: Diagnostic contains: passing @Nullable parameter 'null' where @NonNull is required",
+            "        Foo<Object> f6 = Foo.makeNonNull(null);",
+            "        // BUG: Diagnostic contains: due to mismatched nullability of type parameters",
+            "        Foo<@Nullable Object> f7 = Foo.makeNonNull(new Object());",
+            "        Foo<Object> f8 = Foo.makeNonNull(new Object());",
+            "      }",
+            "    }")
+        .doTest();
+  }
+
+  @Test
+  public void genericInferenceOnAssignmentAfterDeclaration() {
+    makeHelper()
+        .addSourceLines(
+            "Test.java",
+            "package com.uber;",
+            "import org.jspecify.annotations.Nullable;",
+            "    class Test {",
+            "      static class Foo<T extends @Nullable Object> {",
+            "        Foo(T t) {}",
+            "        static <U extends @Nullable Object> Foo<U> makeNull(U u) {",
+            "          return new Foo<>(u);",
+            "        }",
+            "        static <U> Foo<U> makeNonNull(U u) {",
+            "          return new Foo<>(u);",
+            "        }",
+            "      }",
+            "      static void testAssignAfterDeclaration() {",
+            "        // legal",
+            "        Foo<@Nullable Object> f1; f1 = Foo.makeNull(null);",
+            "        // BUG: Diagnostic contains: passing @Nullable parameter 'null' where @NonNull is required",
+            "        Foo<Object> f2; f2 = Foo.makeNull(null);",
+            "      }",
+            "    }")
+        .doTest();
+  }
+
+  @Test
+  public void multipleParametersOneNested() {
+    makeHelper()
+        .addSourceLines(
+            "Test.java",
+            "package com.uber;",
+            "import org.jspecify.annotations.Nullable;",
+            "class Test {",
+            "    static class Foo<T extends @Nullable Object> {",
+            "        Foo(T t) {}",
+            "        static <U extends @Nullable Object> Foo<U> create(U u, Foo<U> other) {",
+            "            return new Foo<>(u);",
+            "        }",
+            "        static void test(Foo<@Nullable Object> f1, Foo<Object> f2) {",
+            "            // no error expected",
```

Comment:
This test currently does not pass.  I would expect no error for the first call to `create`, but an error for the second call to `create` (since the second argument is `Foo<Object>`.  Do you agree?  If so we need to fix this.